addons/stock/procurement.py

```diff
@@ -122,17 +94,16 @@ def _assign_multi(self, cr, uid, procurements, context=None):
                     rule_dict[res_dict[proc]] += [proc]
                 else:
                     rule_dict[res_dict[proc]] = [proc]
-
         for rule in rule_dict.keys():
-            self.write(cr, uid, rule_dict[rule], {'rule_id': rule}, context=context)
-
+            procs = self.browse(rule_dict[rule])
+            procs.write({'rule_id': rule})
 
-    def _get_route_group_dict(self, cr, uid, procurements, context=None):
+    @api.multi
+    def _get_route_group_dict(self):
```

Comment:
Here really need decorator ?? because its calling from already migrated code so check it. :large_blue_circle: